main.go

```diff
@@ -109,22 +115,30 @@ func main() {
 		fmt.Println("err: ", err)
 		return
 	}
+	if len(*apiUrl) != len(*outputFile) {
+		fmt.Println("err: The number of options differs between --api.url and --output.file")
+		return
+	}
 	logger = log.NewSyncLogger(log.NewLogfmtLogger(os.Stdout))
 	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
 
 	ctx := context.Background()
 
 	cfg := sdConfig{
-		ApiUrl:          *apiUrl,
+		OutputFiles:     *outputFile,
+		ApiUrls:         *apiUrl,
 		RefreshInterval: *refreshInterval,
 	}
 
 	disc, err := newDiscovery(cfg)
 	if err != nil {
 		fmt.Println("err: ", err)
 	}
-	sdAdapter := adapter.NewAdapter(ctx, *outputFile, "httpSD", disc, logger)
-	sdAdapter.Run()
+
+	for _, cd := range disc {
+		sdAdapter := adapter.NewAdapter(ctx, cd.outputFile, "httpSD", cd, logger)
+		sdAdapter.Run()
+	}
 
 	<-ctx.Done()
```

Comment:
I think the main function should wait for all adapters are done by using the `sync.WaitGroup`.